src/Sarif.Viewer.VisualStudio.ResultSources.Domain.Core/Services/IResultSourceService.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+
+using CSharpFunctionalExtensions;
+
+using Microsoft.Sarif.Viewer.ResultSources.Domain.Models;
+
+namespace Microsoft.Sarif.Viewer.ResultSources.Domain.Services
+{
+    /// <summary>
+    /// Represents a service that provides analysis results from an external source.
+    /// </summary>
+    public interface IResultSourceService
+    {
+        /// <summary>
+        /// The event raised when new scan results are available.
+        /// </summary>
+        event EventHandler<ResultsUpdatedEventArgs> ResultsUpdated;
+
+        /// <summary>
+        /// Gets the latest code scan results for the current branch.
+        /// </summary>
+        /// <param name="data">A data object.</param>
+        /// <returns>True if the request succeeded; otherwise, an error.</returns>
+        Task<Result<bool, ErrorType>> RequestAnalysisScanResultsAsync(object data = null);
```

Comment:
>data [](http://example.com/codeflow?start=77&length=4)

what is the data parameter? does the type matter? will the it be sent within the request?

Reply:
It can be anything and the implementing service will know its type and what to do with it.